### PR TITLE
Order Filters: apply filters to order list

### PR DESCRIPTION
**Pull request by pmusolino**
As part of #5243 in this PR the filters will be applied to the order list, so we allow merchants to filter by order status and date range.

## Changes
- Removed `orders-load-all-2.json` (no more used. Most of the changes are related to this file)
- Updated tracking event passing also the `date_range` parameter.
- Implemented `computedStartDate` and `computedEndDate` that will return date ranges based on date filter applied.
- Updated `OrderListSyncActionUseCase` removing the dual fetch (because previously we had two tabs `All orders` and `Processing`)
- Implemented the way for refreshing orders when filters are applied.
- Updated/added unit tests.

## Testing
1. Open the order list tab.
2. The order list should be list all the orders without any filter applied.
3. Try to play applying new filters to the order list: status and date ranges.
4. You should be able to see all the orders filtered, and to paginate them.

Update release notes:
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/Classes/ViewRelated/Orders/OrderListSyncActionUseCase.swift [hunk 1]
```diff
@@ -9,42 +9,24 @@ import struct Yosemite.OrderStatus
 ///
 /// ## Discussion
 ///
-/// Pulling to refresh on filtered lists, like the Processing tab, will perform 2 network GET
-/// requests:
+/// Pulling to refresh on filtered lists, will perform 1 network GET
+/// requests with the filters applied (status or date ranges)
```
Comment (jaclync) [inline WooCommerce/Classes/ViewRelated/Orders/OrderListSyncActionUseCase.swift:13]: yay for one less network request! 🎉

File: WooCommerce/Classes/ViewRelated/Orders/OrderListViewController.swift [hunk 2]
```diff
@@ -192,6 +192,10 @@ private extension OrderListViewController {
             self.syncingCoordinator.resynchronize()
         }
 
+        viewModel.onShouldResynchronizeIfNewFitersAreApplied = { [weak self] in
```
Comment (jaclync) [inline WooCommerce/Classes/ViewRelated/Orders/OrderListViewController.swift]: nit:
```suggestion
        viewModel.onShouldResynchronizeIfNewFiltersAreApplied = { [weak self] in
```

File: WooCommerce/Classes/ViewRelated/Orders/Order Filters/Date Range Filter/OrderDateRangeFilter.swift [hunk 3]
```diff
@@ -7,6 +7,49 @@ struct OrderDateRangeFilter: Equatable {
...
+        case .last30Days:
+            return Calendar.current.date(byAdding: .day, value: -30, to: Date())?.startOfDay(timezone: TimeZone.siteTimezone)
+        case .custom:
+            return startDate?.startOfDay(timezone: TimeZone.siteTimezone)
+        default:
+            return nil
+        }
+    }
+
+    var computedEndDate: Date? {
```
Comment (jaclync) [inline WooCommerce/Classes/ViewRelated/Orders/Order Filters/Date Range Filter/OrderDateRangeFilter.swift:28]: these two computed properties feel like could be unit testing candidates? 😄

File: WooCommerce/WooCommerceTests/ViewRelated/Orders/OrderListSyncActionUseCaseTests.swift [hunk 4]
```diff
@@ -162,6 +163,34 @@ final class OrderListSyncActionUseCaseTests: XCTestCase {
...
         XCTAssertEqual(pageSize, self.pageSize)
     }
+
+    // Test that when refresh on a filtered list, the action
+    // returned will be for:
+    //
+    // 1. deleting all orders
+    // 2. fetching all the orders filtered
+    //
+    func test_refresh_with_new_filters_applied_deletes_and_performs_dual_fetch() {
```
Comment (jaclync) [inline WooCommerce/WooCommerceTests/ViewRelated/Orders/OrderListSyncActionUseCaseTests.swift]: nit: I think it's single fetch now?

File: Yosemite/YosemiteTests/Stores/OrderStoreTests+FetchFilteredAndAllOrders.swift [hunk 5]
```diff
@@ -68,11 +68,10 @@ final class OrderStoreTests_FetchFilteredAndAllOrders: XCTestCase {
         XCTAssertEqual(countOrders(), Fixtures.ordersLoadAllJSON.ordersCount + 1)
     }
 
-    func testWhenGivenAFilterItFetchesBothTheFilteredListAndTheAllOrdersList() {
+    func test_when_given_a_filter_it_fetches_all_orders_list() {
```
Comment (jaclync) [inline Yosemite/YosemiteTests/Stores/OrderStoreTests+FetchFilteredAndAllOrders.swift:71]: nit: is this supposed to be fetching filtered orders list?

File: WooCommerce/WooCommerceTests/ViewRelated/Orders/OrderListViewModelTests.swift [hunk 6]
```diff
@@ -396,6 +400,46 @@ final class OrderListViewModelTests: XCTestCase {
...
+        viewModel.activate()
+
+        // Act
+        viewModel.updateFilters(filters: FilterOrderListViewModel.Filters(orderStatus: .completed, dateRange: nil, numberOfActiveFilters: 1))
+
+        // Assert
+        XCTAssertTrue(resynchronizeRequested)
+    }
+
+    func test_given_identical_filters_it_does_not_request_a_resynchronization() {
```
Comment (jaclync) [inline WooCommerce/WooCommerceTests/ViewRelated/Orders/OrderListViewModelTests.swift:424]: nice!

**Review by jaclync — APPROVED** [review]
LGTM! Exciting seeing the order filters working, with way less code and one less network request :shipit: 

I just noticed some issues in the date range picker, unrelated to this PR:

- I can't seem to select the current date in the date picker by tapping on the current date. Tapping on another date then back to the current date works
- If I already selected a range and edit again, then select the end date to be the same as the start date, I can't select a later end date anymore:

https://user-images.githubusercontent.com/1945542/141984077-ef09324c-124e-418e-a972-0520e920d7e9.mov

File: WooCommerce/Classes/ViewRelated/Orders/Order Filters/Date Range Filter/OrderDateRangeFilter.swift (hunk 3 above)
Comment (pmusolino) [reply WooCommerce/Classes/ViewRelated/Orders/Order Filters/Date Range Filter/OrderDateRangeFilter.swift:28]: Implemented here 7dd3ef25e4909d40972f78db32824d866f86ca6a 👍

**Comment by pmusolino** [comment]
@jaclync thank you for your review! 🔢 I added the new issue you discovered in my todo list https://github.com/woocommerce/woocommerce-ios/issues/5243#issuecomment-970299538

**Pull request merged**